### PR TITLE
[5.2] Rename "lists" to "pluck" in the query builder

### PR DESCRIPTION
**Pull request by JosephSilber**
...to be consistent with `Collection@pluck`, `Arr::pluck` and `array_pluck`.

Keeping `lists` as an alias.

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 1]
```diff
@@ -243,6 +243,18 @@ public function lists($column, $key = null)
     }
 
     /**
+     * Alias for the "pluck" method.
+     *
+     * @param  string  $column
+     * @param  string  $key
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Builder.php]: string|null

**Comment by GrahamCampbell** [comment]
Maybe we should mark the old function as deprecated then?

**Comment by crynobone** [comment]
> Maybe we should mark the old function as deprecated then?

I thought we're all about consistent in GB vs Eloquent, Eloquent\Builder::lists() is not deprecated, however Eloquent\Builder::pluck() is at the moment. <https://github.com/laravel/framework/blob/5.1/src/Illuminate/Database/Eloquent/Builder.php#L203>

**Comment by brunogaspar** [comment]
lol

**Comment by GrahamCampbell** [comment]
LOL. We'll get there in the end.

**Comment by taylorotwell** [comment]
@JosephSilber fix Eloquent `pluck` too.

**Comment by vlakoff** [comment]
Wow. Just so that you know, just had an hard time working on an app built for Laravel 5.1, but which should have looked at Laravel 5.2 documentation at some point. It was calling `pluck` thus was getting the first row only.

Changing method roles is really tricky. Please, strongly emphasize this in upgrade guides if you do this again! Cheers.

**Pull request merged**